### PR TITLE
publish workflows: flip Bintray/GitHub packages logic

### PR DESCRIPTION
**Pull request by carlocab**
Fixes publish failures from this not being set. (e.g. #74802, #74803, etc.)

Not sure if this by itself will be enough, but I guess we can find out soon.

### CODE REVIEW

File: .github/workflows/publish-commit-bottles.yml [hunk 1]
```diff
@@ -64,6 +64,7 @@ jobs:
 
       - name: Pull bottles
         env:
+          HOMEBREW_GITHUB_PACKAGES_USER: brewtestbot
```
Comment (MikeMcQuaid) [inline .github/workflows/publish-commit-bottles.yml:67]: This definitely shouldn't be needed here as it should be uploading to Bintray still at this point.
```suggestion
```

**Comment by MikeMcQuaid** [comment]
@carlocab I've pushed what I think this needs to be. If this works as expected: it should be updated in the other workflows too. CC @Bo98 for 👀 💭 too.

**Comment by carlocab** [comment]
Yes, I was contemplating doing what you did too, but since I wasn't certain whether that was actually the right fix I just tried to go for the smallest change that looked like it should resolve https://github.com/Homebrew/homebrew-core/runs/2297523154?check_suite_focus=true#step:10:104

File: .github/workflows/publish-commit-bottles.yml [hunk 2]
```diff
@@ -65,18 +65,18 @@ jobs:
       - name: Pull bottles
         env:
           HOMEBREW_GITHUB_API_TOKEN: ${{secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN}}
-          HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
           HOMEBREW_BINTRAY_USER: brewtestbot
           HOMEBREW_BINTRAY_KEY: ${{secrets.HOMEBREW_BINTRAY_KEY}}
-          BREWTESTBOT_NAME_EMAIL: "BrewTestBot <1589480+BrewTestBot@users.noreply.github.com>"
-        run: brew pr-pull --debug --workflows=tests.yml --committer="$BREWTESTBOT_NAME_EMAIL" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
+        run: brew pr-pull --debug --workflows=tests.yml --no-commit --root-url="https://ghcr.io/v2/homebrew/core" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
```
Comment (carlocab) [inline .github/workflows/publish-commit-bottles.yml:70]: If we're no longer uploading to Bintray, I think we can get rid of this entire block.

**Comment by Bo98** [comment]
Ah the default bottle domain probably flipped logic here. I think we now need to pass `--root-url` for the Bintray upload.

Eventually we should move all root URL handling to `pr-upload` and delete the handling of it at build time - I perhaps should have done that before we migrated.

File: .github/workflows/publish-commit-bottles.yml (hunk 2 above)
Comment (Bo98) [reply .github/workflows/publish-commit-bottles.yml:70]: We still need to upload to Bintray until the next `brew` release.

**Comment by MikeMcQuaid** [comment]
Thanks both! Yeh, we need to keep uploading to Bintray until at least a few days after the next release (when we're confident we have no issues with GitHub Packages).

**Comment by carlocab** [comment]
I think that's the right change. There's also some Bintray upload logic in `linux-dispatch-build-bottle.yml` but there's nothing about GitHub packages there, so I left it alone.

File: .github/workflows/dispatch-build-bottle.yml [hunk 3]
```diff
@@ -145,19 +145,19 @@ jobs:
         env:
           HOMEBREW_BINTRAY_USER: brewtestbot
           HOMEBREW_BINTRAY_KEY: ${{secrets.HOMEBREW_BINTRAY_KEY}}
-          HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
-          BREWTESTBOT_NAME_EMAIL: "BrewTestBot <1589480+BrewTestBot@users.noreply.github.com>"
         run: |
           cd ~/bottles
-          brew pr-upload --verbose --keep-old --committer="$BREWTESTBOT_NAME_EMAIL"
+          brew pr-upload --verbose --keep-old --no-commit --root-url="https://ghcr.io/v2/homebrew/core"
```
Comment (Bo98) [inline .github/workflows/dispatch-build-bottle.yml:150]:
```suggestion
          brew pr-upload --verbose --keep-old --root-url="https://ghcr.io/v2/homebrew/core" --committer="$BREWTESTBOT_NAME_EMAIL"
```

Best to move `--no-commit` to Bintray so it's easier to remove.

File: .github/workflows/dispatch-build-bottle.yml [hunk 4]
```diff
@@ -145,19 +145,19 @@ jobs:
...
       - name: Upload bottles to GitHub Packages
         env:
           HOMEBREW_GITHUB_PACKAGES_USER: brewtestbot
           HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
+          HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
+          BREWTESTBOT_NAME_EMAIL: "BrewTestBot <1589480+BrewTestBot@users.noreply.github.com>"
         run: |
           cd ~/bottles
-          brew pr-upload --verbose --keep-old --no-commit --root-url="https://ghcr.io/v2/homebrew/core"
+          brew pr-upload --verbose --keep-old --committer="$BREWTESTBOT_NAME_EMAIL"
```
Comment (Bo98) [inline .github/workflows/dispatch-build-bottle.yml:160]: Need `--root-url` here. That's what the problem was.

```suggestion
          brew pr-upload --verbose --keep-old --no-commit --root-url="https://homebrew.bintray.com/bottles"
```

File: .github/workflows/dispatch-build-bottle.yml [hunk 5]
```diff
@@ -145,19 +145,19 @@ jobs:
...
           HOMEBREW_BINTRAY_USER: brewtestbot
           HOMEBREW_BINTRAY_KEY: ${{secrets.HOMEBREW_BINTRAY_KEY}}
-          HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
-          BREWTESTBOT_NAME_EMAIL: "BrewTestBot <1589480+BrewTestBot@users.noreply.github.com>"
         run: |
           cd ~/bottles
-          brew pr-upload --verbose --keep-old --committer="$BREWTESTBOT_NAME_EMAIL"
+          brew pr-upload --verbose --keep-old --no-commit --root-url="https://ghcr.io/v2/homebrew/core"
 
       - name: Upload bottles to GitHub Packages
```
Comment (Bo98) [inline .github/workflows/dispatch-build-bottle.yml:152]: Update the names to match:

```suggestion
      - name: Upload bottles to Bintray
```

File: .github/workflows/dispatch-build-bottle.yml (hunk 3 above)
Comment (carlocab) [reply .github/workflows/dispatch-build-bottle.yml:150]: Is that not what I did? This is the part where things are uploaded to Bintray.

Comment (Bo98) [reply .github/workflows/dispatch-build-bottle.yml:150]: `--root-url="https://ghcr.io/v2/homebrew` suggests you are uploading to GitHub packages.

Comment (carlocab) [reply .github/workflows/dispatch-build-bottle.yml:150]: Hmm, ok, I think I see where you're going.

Comment (carlocab) [reply .github/workflows/dispatch-build-bottle.yml:150]: Hmmm, ok, gotcha

File: .github/workflows/dispatch-build-bottle.yml (hunk 4 above)
Comment (carlocab) [reply .github/workflows/dispatch-build-bottle.yml:160]: Should we do the same in `publish-commit-bottles.yml`?

Comment (Bo98) [reply .github/workflows/dispatch-build-bottle.yml]: Yes.

File: .github/workflows/dispatch-build-bottle.yml [hunk 6]
```diff
@@ -141,23 +141,23 @@ jobs:
...
-          HOMEBREW_BINTRAY_USER: brewtestbot
-          HOMEBREW_BINTRAY_KEY: ${{secrets.HOMEBREW_BINTRAY_KEY}}
+          HOMEBREW_GITHUB_PACKAGES_USER: brewtestbot
+          HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
           HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
           BREWTESTBOT_NAME_EMAIL: "BrewTestBot <1589480+BrewTestBot@users.noreply.github.com>"
         run: |
           cd ~/bottles
-          brew pr-upload --verbose --keep-old --committer="$BREWTESTBOT_NAME_EMAIL"
+          brew pr-upload --verbose --keep-old --committer="$BREWTESTBOT_NAME_EMAIL" --root-url="https://ghcr.io/v2/homebrew/core"
```
Comment (carlocab) [inline .github/workflows/dispatch-build-bottle.yml:152]: Setting `--root-url` here is probably unneeded, but I haven't looked at the `pr-upload` code yet to be sure.

Comment (Bo98) [reply .github/workflows/dispatch-build-bottle.yml:152]: It infers it from the default at build time so I'd make it explicit for now (including in `publish-commit-bottles.yml`).

Comment (carlocab) [reply .github/workflows/dispatch-build-bottle.yml:152]: Done.

File: .github/workflows/publish-commit-bottles.yml [hunk 7]
```diff
@@ -62,21 +62,21 @@ jobs:
...
-
       - name: Pull and upload bottles to GitHub Packages
         env:
+          BREWTESTBOT_NAME_EMAIL: "BrewTestBot <1589480+BrewTestBot@users.noreply.github.com>"
+          HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
           HOMEBREW_GITHUB_API_TOKEN: ${{secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN}}
           HOMEBREW_GITHUB_PACKAGES_USER: brewtestbot
           HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
-        run: brew pr-pull --debug --workflows=tests.yml --no-commit --verbose --root-url="https://ghcr.io/v2/homebrew/core" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
+        run: brew pr-pull --debug --workflows=tests.yml --committer="$BREWTESTBOT_NAME_EMAIL" --root-url="https://ghcr.io/v2/homebrew/core"  ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
```
Comment (Bo98) [inline .github/workflows/publish-commit-bottles.yml:72]: Extra space:

```suggestion
        run: brew pr-pull --debug --workflows=tests.yml --committer="$BREWTESTBOT_NAME_EMAIL" --root-url="https://ghcr.io/v2/homebrew/core" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
```

**Review by Bo98 — DISMISSED** [review]

File: .github/workflows/publish-commit-bottles.yml (hunk 7 above)
Comment (carlocab) [reply .github/workflows/publish-commit-bottles.yml:72]: oops

**Review by Bo98 — APPROVED** [review]

**Comment by carlocab** [comment]
Success! https://github.com/Homebrew/homebrew-core/actions/runs/730062208

It's still setting `root_url`, though: 44a4edb21802fa68e00282651aa02f41210b454b

**Comment by Bo98** [comment]
Ah that looks like a bug in our bottle ERB.

**Pull request merged**